gui/wxpython/datacatalog/tree.py

```diff
@@ -893,6 +910,50 @@ def OnSwitchLocationMapset(self, event):
             self.changeLocation.emit(mapset=self.selected_mapset[0].label, location=self.selected_location[0].label)
         self.ExpandCurrentMapset()
 
+    def OnCreateMapset(self, event):
+        """Create new mapset"""
+        self.gisdbase = gisenv()['GISDBASE']
+        self.location = gisenv()['LOCATION_NAME']
+
+        dlg = NewMapsetDialog(
+            parent=self,
+            default=get_default_mapset_name(),
+            database=self.gisdbase,
+            location=self.location
+        )
+        if dlg.ShowModal() == wx.ID_OK:
+            mapset = dlg.GetValue()
+            try:
+                create_mapset(self.gisdbase, self.location, mapset)
+                self.InsertMapset(name=mapset, location_node=self.location,
+                                     mapset_name=mapset)
+                return True
+            except Exception as e:
```

Comment:
It's a good practice to make the exception more specific if you can. For example, put it only around create_mapset and if  you can see what type of exception it throws, it's better to limit it too.

Reply:
Could you give me some example?

Reply:
For type of error, either just simply try it in a python shell if that's simple, or check Python documentation. You can use OSError here.
Please also read e.g. https://realpython.com/python-exceptions/ and https://realpython.com/the-most-diabolical-python-antipattern/ (ignore the logging part), that should clarify the syntax and motivation.